django/core/handlers/base.py

```diff
@@ -246,6 +242,12 @@ def get_response(self, request):
         if not response_is_rendered and callable(getattr(response, 'render', None)):
             response = response.render()
 
+        if (response.status_code == 404):
```

Comment:
Thanks. I'm relatively new to python, (from a long background in C), so that leaks over sometimes. Fortunately, I didn't include any semicolons in these commits! :-)